LiteNetLib4Mirror/Assets/Mirror/Runtime/Transport/LiteNetLib4Mirror/LiteNetLib4MirrorUtils.cs

```diff
@@ -19,13 +19,13 @@ namespace Mirror.LiteNetLib4Mirror
 	public static class LiteNetLib4MirrorUtils
 	{
 		internal static ushort LastForwardedPort;
-		internal static readonly string ApplicationName;
+		public static string SharedKey { get; set; }

```

Comment:
I'd rather have that shared key as a static protected variable in LiteNetLib4MirrorDiscovery that'd be set by a protected virtual method that could be overriden.

Reply:
OK, sure thing. I will implement it as you describe.

Reply:
Actually, because `LiteNetLib4MirrorTransport` currently uses `LiteNetLib4MirrorUtils.ApplicationName` to generate a connection key, `LiteNetLib4MirrorDiscovery` might not be the best place for this, as for the use case this PR addresses, both network discovery and connection need to be independent of using `Application.productName`, and you probably wouldn't want to have `LiteNetLib4MirrorTransport` reference `LiteNetLib4MirrorDiscovery.SharedKey` in its `GetConnectKey` method. That is why I initially went with adding this to `LiteNetLib4MirrorUtils`.

Perhaps this shared key could be a static protected property in the transport instead, with a protected virtual method to set it? Or even just a serialized variable in the transport, in the same way that `authKey` currently is?

Reply:
GetConnectData which uses GetConnectKey is already overridable there.

Reply:
I think `GetConnectKey` itself needs to be overridable, as `ServerStart` calls it directly. Or `LiteNetLib4MirrorTransport` needs a similar protected `ConnectKey` property and overridable setter method (which would match the proposed change to `LiteNetLib4MirrorDiscovery`).

Reply:
Oh, it does call that, then please make that overridable.